pkgs/development/go-modules/generic/default.nix

```diff
@@ -88,6 +92,11 @@ let
         exit 10
       fi
       go mod vendor
+
+      if [ ${vendFlag} == "true" ]; then
+        echo running vend to rewrite vendor folder
+        ${vend}/bin/vend -commands=download,vendor
```

Comment:
I actually prefer the optionalString variant because it does not require every go build to have download `vend`.

Reply:
I think I did this, let me know if my fix isn't what you intended.